src/ptvsd/_vendored/pydevd/pydevd_file_utils.py

```diff
@@ -195,20 +199,38 @@ def set_ide_os(os):
         'UNIX' or 'WINDOWS'
     '''
     global _ide_os
+    global _normcase_from_client
     prev = _ide_os
     if os == 'WIN':  # Apparently PyCharm uses 'WIN' (https://github.com/fabioz/PyDev.Debugger/issues/116)
         os = 'WINDOWS'
 
     assert os in ('WINDOWS', 'UNIX')
 
+    if DEBUG_CLIENT_SERVER_TRANSLATION:
+        print('pydev debugger: client OS: %s' % (os,))
+
+    _normcase_from_client = normcase
+    if os == 'WINDOWS':
+
+        # Client in Windows and server in Unix, we need to normalize the case.
+        if not IS_WINDOWS:
+
+            def _normcase_from_client(filename):
+                return filename.lower()
```

Comment:
Since we're using the standard `normcase` elsewhere, for consistency, perhaps it's better to import `normcase` from `ntpath` / `posixpath` / `macpath` as needed here, depending on the IDE platform? I believe that's what we did in the old implementation in this scenario.

Reply:
I've done some changes to use the same `normcase` implementation everywhere.

Now, just to note, the actual implementation we use is not just `ntpath.normcase` (on Windows) nor do we use the regular `macpath.normcase` nor `posixpath.normcase` implementations either... I'll try to explain a bit the rationale there:

On `linux` it should always be a `no-op` anyways, so, there isn't a great difference from the default implementation (we just skip some type checks), but on mac, the implementation actually defaults to making it all lower (as on windows) because depending on the filesystem that's being used on `mac` it may be case insensitive or not... 

Now, we don't run into as many problems on mac as on windows because most of the problems in windows are actually due to the drive letter (it's strange, but most of the casing problems aren't really due to the casing of the path which is usually correct, but due to the drive letter which is something that each tool appears to give different, whereas there's no drive letter on mac)... this has always been that way in the debugger -- so, maybe something to keep an eye open for and we could think about changing it in the future, but I wouldn't like to change that in this pull request.